spec/lib/efile/nj/nj1040_calculator_spec.rb

```diff
@@ -307,48 +307,51 @@ def over_65_birth_year
   end
 
   describe 'line 15 - state wages' do
-    context 'when no federal w2s' do
-      let(:intake) { create(:state_file_nj_intake, :df_data_minimal) }
+    let(:intake) { create(:state_file_nj_intake) }
 
+    context 'when no state file w2s' do
       it 'sets line 15 to -1 to indicate the sum does not exist' do
         expect(instance.lines[:NJ1040_LINE_15].value).to eq(-1)
       end
     end
 
-    context 'when 2 federal w2s' do
-      let(:intake) { create(:state_file_nj_intake, :df_data_2_w2s) }
-
+    context 'when 2 state file w2s' do
       it 'sets line 15 to the sum of all state wage amounts' do
+        create(:state_file_w2, state_file_intake: intake, state_wages_amount: 12345)
+        create(:state_file_w2, state_file_intake: intake, state_wages_amount: 50000)
+        instance.calculate
         expected_sum = 12345 + 50000
         expect(instance.lines[:NJ1040_LINE_15].value).to eq(expected_sum)
       end
     end
 
-    context 'when many federal w2s' do
-      let(:intake) { create(:state_file_nj_intake, :df_data_many_w2s) }
-
+    context 'when many state file w2s' do
       it 'sets line 15 to the sum of all state wage amounts' do
+        4.times { create(:state_file_w2, state_file_intake: intake, state_wages_amount: 50000) }
+        instance.calculate
         expected_sum = 50000 + 50000 + 50000 + 50000
         expect(instance.lines[:NJ1040_LINE_15].value).to eq(expected_sum)
       end
     end
   end
 
   describe 'line 27 - total income' do
-    let(:intake) { create(:state_file_nj_intake, :df_data_2_w2s) }
+    let(:intake) { create(:state_file_nj_intake) }
+
     it 'sets line 27 to the sum of all state wage amounts' do
-      line_15_w2_wages = 12345 + 50000
-      expect(instance.lines[:NJ1040_LINE_15].value).to eq(line_15_w2_wages)
-      expect(instance.lines[:NJ1040_LINE_27].value).to eq(line_15_w2_wages)
+      allow(instance).to receive(:calculate_line_15).and_return 50000
```

Comment:
yup, the other way to set up the test is to create `state_file_w2`s. i generally prefer to mock calculator methods in tests when they're not the thing that's being explicitly tested, though, in part because it's just easier to understand the test setup. since the calculation for line 15 is tested, i think it would be redundant to recreate the db records just to check that the output for calculate_line_15 gets copied over to line 27, but maybe i'm missing something! happy to tuple about it or just refactor the test if that's the strategy y'all prefer

Reply:
Ah gotcha. I reviewed the business logic for line 27, which appears to be the total income field, so as we add additional income sources, mocking each of them out and validating that `calculate_line_27` sums them all seems appropriate here.

And it looks like we follow a similar approach elsewhere in the spec for mocking out other lines in the calculator. I'm fine with the existing approach!

https://github.com/codeforamerica/vita-min/blob/main/spec/lib/efile/nj/nj1040_calculator_spec.rb#L510